sklearn/metrics/cluster/_supervised.py

```diff
@@ -175,7 +179,7 @@ def contingency_matrix(
         contingency = contingency.tocsr()
         contingency.sum_duplicates()
     else:
-        contingency = contingency.toarray()
+        contingency = xp.asarray(contingency.toarray(), device=device)
```

Comment:
Do you know what dtype the output is for different array input types? Should we specify int dtype, since that is as accurate as we need?